src/views/bootablevolumes/actions/components/EditBootableVolumesModal.tsx

```diff
@@ -17,73 +19,96 @@ import TabModal from '@kubevirt-utils/components/TabModal/TabModal';
 }) => {
   const { t } = useKubevirtTranslation();
+
   const preferencesNames = useMemo(
     () => Object.keys(convertResourceArrayToMap(preferences)).sort((a, b) => a.localeCompare(b)),
     [preferences],
   );
 
-  const instanceTypesToSizesMap = useMemo(
-    () => getInstanceTypesToSizesMap(instanceTypesNames),
-    [instanceTypesNames],
-  );
-
   const initialParams = useMemo(() => {
     const instanceTypeLabel =
       dataSource?.metadata?.labels?.[DEFAULT_INSTANCETYPE_LABEL]?.split('.');
+    const initialCategory = Object.entries(categoryDetailsMap).find(
+      (category) => category[1].prefix === instanceTypeLabel?.[0],
+    );
 
     return {
       preference: dataSource?.metadata?.labels?.[DEFAULT_PREFERENCE_LABEL],
-      instanceType: instanceTypeLabel?.[0],
-      size: instanceTypeLabel?.[1],
+      instanceType: initialCategory,
+      size: initialCategory && instanceTypeLabel?.[1],
       description: dataSource?.metadata?.annotations?.[ANNOTATIONS.description],
     };
   }, [dataSource]);
 
   const [preference, setPreference] = useState<string>(initialParams.preference);
-  const [instanceType, setInstanceType] = useState<string>(initialParams.instanceType);
+  const [isInstanceTypeOpen, setIsInstanceTypeOpen] = useState(false);
+  const [instanceType, setInstanceType] = useState<string>(initialParams.instanceType?.[0]);
+  const [isSizeOpen, setIsSizeOpen] = useState(false);
   const [size, setSize] = useState<string>(initialParams.size);
   const [description, setDescription] = useState<string>(initialParams.description);
 
-  const sizeOptions = useMemo(
-    () => instanceTypesToSizesMap[instanceType] || [],
-    [instanceType, instanceTypesToSizesMap],
+  // update options for 'Size' dropdown according to chosen 'instanceType' in 'Default InstanceType' dropdown
+  const { instanceTypes }: CategoryDetails = useMemo(
+    () => (instanceType ? categoryDetailsMap[instanceType] : {}),
+    [instanceType],
   );
 
-  const onInstanceTypeChange = useCallback(
-    (instanceTypeName: string) => {
-      setInstanceType(instanceTypeName);
-      setSize(instanceTypesToSizesMap[instanceTypeName][0]);
-    },
-    [instanceTypesToSizesMap],
-  );
+  const onInstanceTypeSelect = (
+    _event: ChangeEvent<HTMLSelectElement>,
+    newInstanceType: string,
+  ) => {
+    setInstanceType(newInstanceType);
+    setIsInstanceTypeOpen(false);
 
-  const onChangeVolumeParams = useCallback(() => {
+    const newCategoryObject = categoryDetailsMap[newInstanceType];
+    const newCategorySize = newCategoryObject.instanceTypes[0].label;
+    setSize(newCategorySize);
+  };
+
+  const onSizeSelect = (_event: ChangeEvent<HTMLSelectElement>, newSize: string) => {
+    setSize(newSize);
+    setIsSizeOpen(false);
+  };
+
+  const onSubmitVolumeParams = useCallback(() => {
     const preferenceLabel = preference && { [DEFAULT_PREFERENCE_LABEL]: preference };
+
+    const categoryObject = categoryDetailsMap[instanceType];
```

Comment:
Using `Object` at the end of the sentence usually doesn't communicate much useful information. It's better to just use a name that helps to identify what type of data that object holds. In this case, this object is an instance of a type, `CategoryDetails`, so it makes since to use the type name as the variable name.